src/include/storage/checkpoint_manager.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include "common/spin_latch.h"
+#include "common/strong_typedef.h"
+#include "storage/checkpoint_io.h"
+#include "storage/projected_columns.h"
+#include "storage/sql_table.h"
+#include "storage/write_ahead_log/log_io.h"
+#include "transaction/transaction_context.h"
+#include "transaction/transaction_defs.h"
+
+namespace terrier::storage {
+
+/**
+ * A CheckpointManager is responsible for serializing tuples of all tables in the database out and
+ * divide time into epochs by doing checkpoint.
+ */
+class CheckpointManager {
+ public:
+  /**
+   * Constructs a new CheckpointManager, writing its records out to the given file.
+   */
+  explicit CheckpointManager(const char *log_file_path_prefix) : log_file_path_prefix_(log_file_path_prefix) {}
+
+  /**
+   * Start a new checkpoint with the given transaxtion context.
+   * @param txn transaction context this checkpoint will be running under.
+   */
+  void StartCheckpoint(transaction::TransactionContext *txn) {
+    txn_ = txn;
+    out_.Open((GetCheckpointFilePath(txn)).c_str());
```

Comment:
This might be a good place to call Checkpoint on each SqlTable, since this is where the checkpoint it starting.

Reply:
Yes, this is the ultimate goal. However, we do not know anything about existing tables, because there is no working catalog yet.